### PR TITLE
MSF-13: Ability to specify a default tenant for outgoing requests

### PR DESCRIPTION
**Pull request by erdody**
Requested by chatgrid, so they don't have to inject a special header on all their clients.

### CODE REVIEW

**Comment by yixiaol-m** [comment]
Hi

File: linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/EnvTenantHostIdentifierConfig.scala [hunk 1]
```diff
@@ -18,10 +18,11 @@ object EnvTenantHostIdentifierConfig {
 }
 
 class EnvTenantHostIdentifierConfig extends HttpIdentifierConfig {
+  val defaultTenant: Option[String] = None
```
Comment (yixiaol-m) [inline linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/EnvTenantHostIdentifierConfig.scala:21]: Could this be moved to line 26?

File: router/http/src/main/scala/io/buoyant/router/http/EnvTenantHostIdentifier.scala [hunk 2]
```diff
@@ -38,7 +42,7 @@ case class EnvTenantHostIdentifier(
     Dst.Path(prefix ++ path, baseDtab(), Dtab.local)
 
   def apply(req: Request): Future[RequestIdentification[Request]] = {
-    val tenant = getHeader(req, TenantHeader)
+    val tenant = getHeader(req, TenantHeader).orElse(defaultTenant)
```
Comment (yixiaol-m) [inline router/http/src/main/scala/io/buoyant/router/http/EnvTenantHostIdentifier.scala:45]: do we need to do a check here and throw error if both TenantHeader and defaultTenant are not set?

Comment (erdody) [reply router/http/src/main/scala/io/buoyant/router/http/EnvTenantHostIdentifier.scala:45]: No, this is an Option (similar to Java's Optional). We throw below if it's empty.

File: linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/EnvTenantHostIdentifierConfig.scala (hunk 1 above)
Comment (erdody) [reply linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/EnvTenantHostIdentifierConfig.scala:21]: No, this is an attribute of the class. Line 26 would be an argument in the newIdentifier method, which we override.

**Review by abhinigam — APPROVED** [review]
LGTM.

**Review by yixiaol-m — APPROVED** [review]
LGTM

**Pull request merged**